### PR TITLE
Crash on wrong file using Palette Tool

### PR DESCRIPTION
**Pull request by thekolian1996**
Ticket #1890450.

### CODE REVIEW

File: Pinta.Core/Classes/Palette.cs [hunk 1]
```diff
@@ -159,15 +161,44 @@ public void LoadDefault ()
...
+			string secondaryText = string.Format(Catalog.GetString("Could not open file: {0}"), filename);
+			secondaryText += string.Format(Catalog.GetString($"{Environment.NewLine}{Environment.NewLine}Pinta supports the following palette formats:{Environment.NewLine}"));
+			var extensions = from format in PintaCore.System.PaletteFormats.Formats
+							 where format.Loader != null
+							 from extension in format.Extensions
+							 where char.IsLower(extension.FirstOrDefault())
+							 orderby extension
+							 select extension;
+
+			foreach (var extension in extensions)
```
Comment (cameronwhite) [inline Pinta.Core/Classes/Palette.cs:196]: Can `String.Join()` be used here?

File: Pinta.Core/Classes/Palette.cs [hunk 2]
```diff
@@ -159,15 +161,44 @@ public void LoadDefault ()
...
-			colors.TrimExcess ();
-			OnPaletteChanged ();
+			try
+			{
+				var loader = PintaCore.System.PaletteFormats.GetLoaderByFilename (fileName);
+				colors = loader.Load (fileName);
+				colors.TrimExcess ();
+				OnPaletteChanged ();
+			}
+			catch (NullReferenceException e)
```
Comment (cameronwhite) [inline Pinta.Core/Classes/Palette.cs:171]: Maybe `GetLoaderByFilename()` should be changed to just return `null` if the format doesn't exist, similar to `ImageFormats.GetImporterByFile()`? It's a bit weird that it throws a NullReferenceException there

**Review by cameronwhite — COMMENTED** [review]
Looks good, just a couple suggestions in the comments!

File: Pinta.Core/Classes/Palette.cs (hunk 1 above)
Comment (thekolian1996) [reply Pinta.Core/Classes/Palette.cs]: Wow, that was stupid. I replaced it with ```String.Join()``` here and in my first pull request about unsupported file format.

File: Pinta.Core/Classes/Palette.cs (hunk 2 above)
Comment (thekolian1996) [reply Pinta.Core/Classes/Palette.cs]: Done. Changed the code to the style similar to ```WorkspaceManager.OpenFile()``` method. I hope it is correct this time.

**Review by cameronwhite — APPROVED** [review]
Looks good, thanks!

**Pull request merged**